PsfLauncher/main.cpp

```diff
@@ -125,53 +122,109 @@ int launcher_main(PWSTR args, int cmdShow) noexcept try
+        startupInfo.dwFlags = STARTF_USESHOWWINDOW;
+        startupInfo.wShowWindow = static_cast<WORD>(cmdShow);
 
-    // Propagate exit code to caller, in case they care
-    switch (::WaitForSingleObject(processInfo.hProcess, INFINITE))
-    {
-    case WAIT_OBJECT_0:
-        break; // Success case... handle at the end of main
+        Log("\tCreating process %ls", cmdLine.data());
+        // NOTE: PsfRuntime, which we dynamically link against, detours this call
+        PROCESS_INFORMATION processInfo;
+        if (!::CreateProcessW(
+            exePath.c_str(),
+            cmdLine.data(),
+            nullptr, nullptr, // Process/ThreadAttributes
+            true, // InheritHandles
+            0, // CreationFlags
+            nullptr, // Environment
+            dirStr ? (packageRoot / dirStr).c_str() : nullptr, // NOTE: extended-length path not supported
+            &startupInfo,
+            &processInfo))
+        {
+            std::wostringstream ss;
+            auto err{ ::GetLastError() };
+            // Remove the ".\r\n" that gets added to all messages
+            auto msg = widen(std::system_category().message(err));
+            msg.resize(msg.length() - 3);
+            ss << L"ERROR: Failed to create detoured process\n  Path: \"" << exeName << "\"\n  Error: " << msg << " (" << err << ")";
+            ::PSFReportError(ss.str().c_str());
+            return err;
+        }
+
+        // Propagate exit code to caller, in case they care
+        switch (::WaitForSingleObject(processInfo.hProcess, INFINITE))
+        {
+        case WAIT_OBJECT_0:
+            break; // Success case... handle at the end of main
 
-    case WAIT_FAILED:
-        return ::GetLastError();
+        case WAIT_FAILED:
+            return ::GetLastError();
 
-    default:
-        return ERROR_INVALID_HANDLE;
-    }
+        default:
+            return ERROR_INVALID_HANDLE;
+        }
 
-    DWORD exitCode;
-    if (!::GetExitCodeProcess(processInfo.hProcess, &exitCode))
+        DWORD exitCode;
+        if (!::GetExitCodeProcess(processInfo.hProcess, &exitCode))
+        {
+            return ::GetLastError();
+        }
+        return exitCode;
+    }
+    else
     {
-        return ::GetLastError();
+        // Non Exe case, use shell launching to pick up local FTA
+        auto nonexePath = packageRoot / exeName;
+
+        CoInitializeEx(NULL, COINIT_APARTMENTTHREADED | COINIT_DISABLE_OLE1DDE);
+        SHELLEXECUTEINFO shex;
+        memset(&shex, 0, sizeof(SHELLEXECUTEINFO));
+        shex.cbSize = sizeof(shex);
+        shex.fMask = SEE_MASK_NOCLOSEPROCESS;
+        shex.hwnd = (HWND)NULL;
+        shex.lpVerb = NULL;  // just use the default action for the fta
+        shex.lpFile = nonexePath.c_str();
+        shex.lpParameters = exeArgString;
+        shex.lpDirectory = dirStr ? (packageRoot / dirStr).c_str() : nullptr;
+        shex.nShow = static_cast<WORD>(cmdShow);
+        //shex.hInstApp = 0; // for output

```

Comment:
Did you leave these commented lines in here for a reason?  I see you say "for output".  Should these lines be un-commented for a specific reason?

Reply:
No. Lines were replaced by memset.  Removed commented lines.